sdks/java/build-tools/src/main/resources/beam/findbugs-filter.xml

```diff
@@ -139,6 +139,33 @@
     <!-- Takes ownership of input buffer -->
   </Match>
 
+  <Match>
+    <Class name="org.apache.beam.runners.dataflow.util.MonitoringUtilTest" />
+    <Field name="thrown" />
+    <Bug pattern="URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD" />
```

Comment:
We can now localize `@SuppressFBWarnings` to the location of the warning, via our ASL2.0 licensed annotation package. Incidentally I learned afterwards IIRC that there's already a findbugs `@SuppressWarnings` annotation that you can use if you fully qualify it, thus not even requiring the `FB`.

Reply:
But aren't we trying also in the long term to get rid of findbugs? If this is the goal, better do this in the xml and we get of it and that's it, no?

Reply:
Can you elaborate on what has changed? Here's my understanding of things:

* FindBugs uses a [`@SuppressFBWarnings`](http://findbugs.sourceforge.net/api/edu/umd/cs/findbugs/annotations/SuppressFBWarnings.html) annotation for inline suppressions.
* There is also a [`@edu.umd.cs.findbugs.annotations.SuppressWarnings`](http://findbugs.sourceforge.net/api/edu/umd/cs/findbugs/annotations/SuppressWarnings.html), but it is marked as deprecated in favor of `@SuppressFBWarnings` since it clashes the other `@SuppressWarnings` annotation
* We [previously removed references](https://github.com/apache/beam/pull/1401) to `@SuppressFBWarnings` because it is only available via [`com.google.code.findbugs:annotations](https://mvnrepository.com/artifact/com.google.code.findbugs/annotations) under LGPL.

Also it seems the general direction is to stop supporting FindBugs in favor of ErrorProne / SpotBugs. So it doesn't seem worth much to invest in migrating the suppressions file.

Reply:
Ping @kennknowles; can you help me understand your feedback above?

Reply:
Ping @kennknowles. 

Let me know if you have additional comments. If I don't hear back I'll look for a different reviewer.

Reply:
There's actually an ASF licensed compatible annotation package which we depend on already. But I'd rather just get this in as-is. The XML is fine too, though then we will just delete it I guess and might miss some places where the new tool isn't protecting us.

Reply:
I see in [`build_rules.gradle`](https://github.com/apache/beam/blob/6a327f222bb6f7fedb694405fc945caf4671dbbb/build_rules.gradle#L196) that we have a `findbugs_annotations` dependency which looks new. It points to [`com.google.code.findbugs:findbugs-annotations:3.0.1`](https://mvnrepository.com/artifact/com.google.code.findbugs/findbugs-annotations/3.0.1). Looking in the jar I see the `@SuppressFBWarmings` annotation, so this might work, although it's also listed as LGPL. Is LGPL ok to use?

I don't see any existing usages of `@SuppressFBWarnings` in our code base, which is why I didn't tackle it here. As you mentioned above, let's save it for another day.

Reply:
Excellent point @swegner, it seems that as part of the migration to gradle we forgot to use the correct library for findbugs-annotations, we should be using this one https://github.com/stephenc/findbugs-annotations as the maven build does.
But gradle shows the one you mention and we cannot add LGPL code for this, so this is a release blocker.

Reply:
I am trying to remove the GPL dependency in https://github.com/apache/beam/pull/5609. It's broken the build, so I will investigate.

Reply:
@iemejia @pabloem Thanks for following-up on this!